sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java

```diff
@@ -59,14 +60,22 @@
     private static final BinaryData EMPTY_DATA = new BinaryData(new byte[0]);
     private static final int STREAM_READ_SIZE = 1024;
 
-    private static final Object LOCK = new Object();
-
-    private static volatile JsonSerializer defaultJsonSerializer;
+    private static final JsonSerializer SERIALIZER;
+    static {
+        JsonSerializer tmp;
+        try {
+            tmp = JsonSerializerProviders.createInstance();
+        } catch (IllegalStateException e) {
+            tmp = new JacksonSerializer();
+        }
+        SERIALIZER = tmp;
```

Comment:
Could we turn this into a utility method somewhere in implementation, I think this gets used in a couple other spots.

Reply:
@alzimmermsft, @srnagar, @JonathanGiles   How do you like my suggestion to have `JsonSerializer.getDefaultSerializer()` in https://github.com/Azure/azure-sdk-for-java/pull/19857#issuecomment-799664819?

It has two benefits as listed below but you might have other considerations I neglect.
1. Having a fallback json serializer in the implementation of `azure-core` would allow `CloudEvent`, 'BinaryData`, and other classes in `azure-core` to use it. But other SDK classes like `EventGridEvent` doesn't have access to it. This statice method would allow `EventGridEvent` to access the fallback json serializer`.
2. `CloudEvent`, `BinaryData` and other classes all need to have their own code to initialize this default serializer. This new static method will improve code reuse because they can simply call this method.

Reply:
Updated the design to add a new API in `JsonSerializerProviders` to support a fallback-implementation if required.